src/main/java/com/demo/jwt/JwtMybatisApplication/controller/TeacherController.java

```diff
@@ -25,22 +26,26 @@ public class TeacherController {
     TeacherService teacherService;
 
     @PostMapping
+    @RolesAllowed("ROLE_ADMIN")
     public TeacherEntity addTeacher(@RequestBody TeacherAdditionDto teacherAdditionDto){
         return teacherService.addTeacher(teacherAdditionDto);
     }
 
     @GetMapping("/{id}")
+    @RolesAllowed({"ROLE_ADMIN","ROLE_TEACHER"})
     public TeacherDisplayDto getTeacherById(@PathVariable long id){
         return teacherService.getTeacherById(id);
     }
 
     @GetMapping
+    @RolesAllowed({"ROLE_ADMIN","ROLE_TEACHER"})
     public List<TeacherDisplayInfoDto> getAllTeachers(){
         return teacherService.getAllTeachers();
     }
 
     @GetMapping("/{teacherId}/subject")
```

Comment:
```suggestion
    @GetMapping("/{teacherId}/subjects")
```
Keep the singular/plural naming convention consistent

Reply:
Done